src/AzureDeprecation.APIs.REST/DataAccess/Repositories/CosmosDbDeprecationsRepository.cs

```diff
@@ -0,0 +1,52 @@
+﻿using AzureDeprecation.APIs.REST.DataAccess.Contracts;
+using AzureDeprecation.APIs.REST.DataAccess.Models;
+using AzureDeprecation.APIs.REST.DataAccess.Utils;
+using AzureDeprecation.APIs.REST.Settings;
+using CodeJam;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Azure.Cosmos.Linq;
+using Microsoft.Extensions.Options;
+
+namespace AzureDeprecation.APIs.REST.DataAccess.Repositories;
+
+internal class CosmosDbDeprecationsRepository : IDeprecationsRepository
+{
+    readonly CosmosClient _client;
+    readonly CosmosDbOptions _dbOptions;
+    
+    public CosmosDbDeprecationsRepository(IOptions<CosmosDbOptions> dbSettings)
+    {
+        Code.NotNull(dbSettings?.Value, nameof(dbSettings));
+        _dbOptions = dbSettings.Value;
+        _client = new CosmosClient(dbSettings.Value.ConnectionString);
+    }
+
+    public async Task<DeprecationNoticesResult> GetDeprecationsAsync(
+        DeprecationsRequestModel deprecationsRequestModel, CancellationToken cancellation = default)
+    {
+        var db = _client.GetDatabase(_dbOptions.DbName);
+        var container = db.GetContainer(_dbOptions.ContainerId);
```

Comment:
What if it does not exist?

Reply:
Added code to check wheteher collection is exists